modules/oe_translation_cdt/src/Model/Transaction.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_translation_cdt\Model;
+
+use Symfony\Component\Serializer\Attribute\SerializedPath;
+
+/**
+ * Defines the DTO class for transaction.
+ */
+class Transaction {
+
+  #[SerializedPath('[TransactionHeader][SenderDetails][@drupalVersion]')]
+  /**
+   * The version of Drupal core.
+   */
+  protected string $drupalVersion;
+
+  #[SerializedPath('[TransactionHeader][SenderDetails][@moduleVersion]')]
+  /**
+   * The version of the module.
+   */
+  protected string $moduleVersion;
+
+  #[SerializedPath('[TransactionHeader][SenderDetails][ProducerDateTime]')]
+  /**
+   * The date of the file.
+   *
+   * @var \DateTimeInterface
+   */
+  protected \DateTimeInterface $producerDateTime;
+
+  #[SerializedPath('[TransactionIdentifier]')]
+  /**
+   * The transaction ID.
+   */
+  protected string $transactionId;
+
+  #[SerializedPath('[TransactionCode]')]
+  /**
+   * The transaction code.
+   */
+  protected string $transactionCode;
+
+  #[SerializedPath('[TransactionData][TranslationDetails][Translation]')]
+  /**
+   * The language items to translate.
+   *
+   * @var \Drupal\oe_translation_cdt\Model\TransactionItem[]
+   */
+  protected array $transactionItems;
+
+  #[SerializedPath('[TransactionData][TranslationDetails][TotalCharacterLength]')]
+  /**
+   * The total character length.
+   */
+  protected int $totalCharacterLength;
+
+  /**
+   * Gets the Drupal version.
+   *
+   * @return string
+   *   The Drupal version.
+   */
+  public function getDrupalVersion(): string {
+    return $this->drupalVersion;
+  }
+
+  /**
+   * Sets the Drupal version.
+   *
+   * @param string $drupalVersion
+   *   The Drupal version.
+   */
+  public function setDrupalVersion(string $drupalVersion): self {
```

Comment:
the convention is variable names are snake case $drupal_version and class attributes yes are lower camel case so $this->drupalVersion

Reply:
You are right, I got too comfortable with PHPStorm getter/setter generation. Will fix.
It is not reported by CoDeSniffer because it is only forbidden to mix cases in one file.